### PR TITLE
Show more possible problems with build_and_check in file paddle_build.sh

### PR DESCRIPTION
**Pull request by Avin0323**
### PR types
Others

### PR changes
Others

### Describe
Since `check_style` and `example` are blocked, `example` would't to run when there are errors in `check_style`. Then whether there are errors in `example`, only display errors in `check_style`. Resulting in multiple commits to fix all problems.

The way is as follows:

1. Function `check_style` and `example` run via `$()`, then `exit` does not take effect in the CI. But `check_style_code` and `example_code` will catch exit status codes for final process.

image
image
image
image

2. Before `assert_api_spec_approvals`, print information and exit in `summary_check_problems` function accroding to `check_style_code` and `example_code`.

image

### CODE REVIEW

**Comment by luotao1** [comment]
因为build_and_check只影响PR_CI_CPU_Py2，因此可以用test=document_fix来跳过其他流水线的CI

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -1407,12 +1407,18 @@ function main() {
...
         cmake_gen_and_build ${PYTHON_ABI:-""} ${parallel_number}
         check_sequence_op_unittest
         generate_api_spec ${PYTHON_ABI:-""} "PR"
-        example
+        $(example >&2)
+        example_code=$?
+        [ $check_style_code -ne 0 ] && echo "Please check the log and fix code style problems."
+        [ $example_code -ne 0 ] && echo "Please check the log and fix example problems."
+        [ $check_style_code -ne 0 ] && exit $check_style_code
+        [ $example_code -ne 0 ] && exit $example_code
```
Comment (lelelelelez) [inline paddle/scripts/paddle_build.sh]: 这样实现没有问题，且不会影响退出码。
但最后的错误可否以下面这种方式提醒？担心只是打印出来，大家会不仔细看。
\\==================
sumary problems:
\\==================
1. check code style failed！ Please check the log and fix code style problems.
2. example code failed！Please check the log and fix example problems.
@luotao1 看是否有必要？

Comment (luotao1) [reply paddle/scripts/paddle_build.sh]: 需要做一个summary的总结

**Review by lelelelelez — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**